web/src/hooks/useLogs.ts

```diff
@@ -479,6 +530,69 @@ export const useLogs = (
     }
   };
 
+  const getVolume = async ({
+    query,
+    tenant,
+    timeRange,
+    namespace,
+  }: {
+    query: string;
+    tenant?: string;
+    timeRange?: TimeRange;
+    namespace?: string;
+  }) => {
+    if (query.length === 0) {
+      dispatch({ type: 'volumeError', payload: { error: new Error('Query is empty') } });
+      return;
+    }
+
+    // Throttle requests
+    if (lastExecutionTime.current.volume && Date.now() - lastExecutionTime.current.volume < 50) {
+      return;
+    }
+
+    try {
+      currentQuery.current = query;
+      currentTenant.current = tenant ?? currentTenant.current;
+      currentTime.current = Date.now();
+      lastExecutionTime.current.logs = Date.now();
+      currentTimeRange.current = timeRange ?? currentTimeRange.current;
+
+      const { start, end } = numericTimeRange(currentTimeRange.current);
+
+      dispatch({ type: 'volumeRequest' });
+
+      if (volumeAbort.current) {
+        volumeAbort.current();
+      }
+
+      await fetchConfig();
+
+      // Volume API only accepts labels, so have to extract them from the query.
+      // Only grabs the data within the { }
+      query = query.match(/\{.*\}/g)?.[0] ?? query;
```

Comment:
Using the LogQL class might be better as it uses a lezer to analyze and parse the query, this regex might not work in all contexts, for example complex queries.